Feature/article like api

taejunUM opened the pull request:
## 📄 게시글 좋아요 기능 구현
<!-- ex. feat: 회원가입 API 구현 -->

#### 📎 관련 이슈
<!-- 연결된 이슈 번호 -->
close #28 

## ✅ 변경 사항
<!-- 어떤 작업을 했는지 간략히 정리 -->
- 게시글 좋아요 (toogle) / likeCount, likedMember 조회 기능 구현
- 서비스 로직 내 중복 검증 로직 메서드 분리 (member, article, clubMember)

## 📸 화면 캡처 (선택)
<!-- UI 변경이 있다면 스크린샷 첨부 -->

## 🔍 테스트 방법
<!-- 어떤 방식으로 동작을 검증했는지 -->
- [x] 로컬에서 직접 테스트
- [ ] 유닛 테스트 작성 및 통과
- [ ] 브라우저 / 앱 화면에서 확인

## 👀 중점적으로 봐줬으면 하는 부분
<!-- 리뷰어가 집중해서 봐야 할 부분이 있다면 -->

## 📝 기타 사항
<!-- 추가적으로 공유할 내용 -->
- member, article, clubMember 검증로직 추후 리팩토링
- 추후 queryDSL 변경 고려

taejunUM committed:
refactor: 기능별like 패키지 통합

refactor: 기능별like 패키지 통합

taejunUM committed:
feat: 테스트용 멤버 주입 메서드 구현

taejunUM committed:
feat: LikeController 구현

taejunUM committed:
feat: like&unlike 토글 반환 dto 구현

taejunUM committed:
feat: ArticleLikeRepository 구현

taejunUM committed:
feat: like&unlike 토글 서비스로직 구현

- 멤버 검증 로직은 ClubMember 구현후 리팩토링 (TODO작성)

taejunUM committed:
feat: 좋아요 갯수 반환 dto 구현

taejunUM committed:
feat: controller 좋아요 갯수, 멤버 반환 api 구현

taejunUM committed:
feat: 좋아요 멤버 반환 dto 구현

taejunUM committed:
refactor: like & unlike dto 수정

taejunUM committed:
feat: ArticleLike Repository 기능 추가

taejunUM committed:
feat: ArticleService 갯수, 누른멤버 반환 구현

taejunUM committed:
refactor: ArticleLike 중복 검증 로직 통합

taejunUM committed:
refactor: ArticleLike변수명 변경

taejunUM committed:
refactor: ArticleLike 멤버생성용 임시 메서드 삭제

taejunUM committed:
style: articleLike 줄바꿈, 들여쓰기

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/like/articleLike/dto/response/ArticleLikeCountResponseDto.java:
```diff
@@ -0,0 +1,7 @@
+
+public record ArticleLikeCountResponseDto(Long articleId, int likeCount) {
+    public static ArticleLikeCountResponseDto toDto(Long articleId, int count) {
+        return new ArticleLikeCountResponseDto(articleId, count);
```
`toDto` 메서드와 `new ArticleLikeCountResponseDto` 생성자의 두번째 파라미터 네이밍을 다르게 설정한 이유가 있을까요?

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/like/articleLike/dto/response/ArticleLikeToggleResponseDto.java:
```diff
@@ -0,0 +1,21 @@
+                articleId.getId(),
+                memberId.getId(),
+                isLiked ? "LIKED" : "UNLIKED"
+        );
```
파라미터 네이밍은 articleId -> article, memberId -> member 로 수정하는게 좋아보입니다
`ArticleLikeToggleResponseDto` 의 생성자와 toDto 메서드의 파라미터 순서가 틀려서 코드를 읽는데 불편함이 있습니다 2개의 순서는 맞추는게 좋아보여요:)

taejunUM replied:
네이밍, 순서 수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/like/articleLike/service/ArticleLikeService.java:
```diff
@@ -0,0 +1,78 @@
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class ArticleLikeService {
```
한칸의 공백이 있는게 컨벤션에 맞을것같아요

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/like/articleLike/service/ArticleLikeService.java:
```diff
@@ -0,0 +1,78 @@
+        Article article = findArticleOrThrow(articleId);
+        int count = articleLikeRepository.countByArticle(article);
+
+        return ArticleLikeCountResponseDto.toDto(articleId, count);
```
찾아온 article 객체를 주입하고 dto 내부에서 .getId 로 아이디 값을 주입하는게 조금 더 적절해 보입니다

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/like/articleLike/controller/ArticleLikeController.java:
```diff
@@ -0,0 +1,46 @@
+    ) {
+        return ResponseEntity.status(HttpStatus.OK)
+                .body(articleLikeService.getArticleLikeCount(articleId));
+    }
```
기획상 Article 을 조회할 때 Like 데이터도 함께 조회가 됩니다. 컨트롤러에는 toggle api 를 제외한 나머지 api 는 필요하지 않을것같습니다:)

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/like/articleLike/dto/response/ArticleLikeMemberResponseDto.java:
```diff
@@ -0,0 +1,9 @@
+    public static ArticleLikeMemberResponseDto toDto(Member member) {
+        return new ArticleLikeMemberResponseDto(member.getId(), member.getName(), member.getProfileImage());
+    }
+}
```
현재 Dto 중 `ArticleLikeToggleResponseDto` 를 제외한 2가지 Dto 는 구조 변경이 필요해 보입니다.
기획상 좋아요의 갯수와 좋아요를 누른 멤버를 모두 보여줘야하지만 Dto 가 분리되어 있어, 갯수조회 api 호출과 멤버 조회 api 호출이 따로 이루어지게 됩니다. 최상위 Dto 에서 좋아요 갯수를, 하위 Dto 에서 좋아요를 누른 Member 리스트를 가지고 있다면 하나의 Dto 로 커버가 가능할 것 같습니다.

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/like/articleLike/repository/ArticleLikeRepository.java:
```diff
@@ -0,0 +1,18 @@
+    int countByArticle(Article article);
+
+    List<ArticleLike> findAllByArticle(Article article);
+}
```
메서드 네이밍이 메서드가 어떤걸 변화시키는지 명확하지 않습니다.
Spring Data JPA 메서드 네이밍 관례를 찾아보는걸 추천합니다:)

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/like/articleLike/controller/ArticleLikeController.java:
```diff
@@ -0,0 +1,46 @@
+        return ResponseEntity.status(HttpStatus.OK)
+                .body(articleLikeService.getArticleLikedMembers(articleId));
+    }
+}
```
전체적인 컨트롤러 메서드의 네이밍이 명확하지 않은 것 같습니다 수정이 필요해 보여요:)

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/like/articleLike/entity/ArticleLike.java:
```diff
@@ -1,4 +1,4 @@
+package com.mobble.mobbleserver.like.articleLike.entity;
 
 import com.mobble.mobbleserver.domain.article.entity.Article;
 import com.mobble.mobbleserver.domain.member.entity.Member;
```
like 패키지를 domain 패키지 밖으로 분리한 이유가 있을까요?

taejunUM replied:
패키지 리팩토링중 실수가 있었나봅니다. 수정 완료!

taejunUM committed:
refactor: like 패키지 경로 변경

taejunUM committed:
refactor: articleLike 컨트롤러 api 제거

taejunUM committed:
articleLike 컨트롤러 메서드 네이밍 수정

taejunUM committed:
articleLike 파라미터 네이밍 변경

taejunUM committed:
refactor: dto 네이밍, 순서 및 파라미터 타입 변경

taejunUM committed:
style: articleLike 공백 추가

Developer-Groo committed:
chore: 공백제거

Developer-Groo committed:
style: 컨벤션에 따른 파라미터 줄바꿈

taejunUM committed:
refactor: articleLike JPA메서드 네이밍 수정

taejunUM committed:
refactor: Dto 구조 변경

taejunUM committed:
refactor: articleLike repository 파라미터 타입 변경

taejunUM committed:
articleLike dto구조 변경, service 메서드 통합

taejunUM committed:
style: articleLike dto 줄바꿈

Developer-Groo committed:
refactor: messages.properties 파일 제거

Developer-Groo committed:
feat: ErrorCode interface 구현

Developer-Groo committed:
feat: DomainException 구현

Developer-Groo committed:
feat: ErrorResponseDto 구현 및 패키지 변경

Developer-Groo committed:
feat: CommentErrorCode 구현

Developer-Groo committed:
feat: CommentValidationErrorCode 구현

Developer-Groo committed:
feat: Comment entity 검증 로직 구현 및 커스텀 에러 적용

Developer-Groo committed:
feat: findByIdAndMemberId 메서드 반환 타입 Optional 로 변경

Developer-Groo committed:
feat: CommentValidator 계층 추가

- 반복되는 검증로직 분리
- 도메인 별 검증로직 통일화

Developer-Groo committed:
refactor: 기존 검증 로직 Validator 로 변경

Developer-Groo committed:
feat: DomainExceptionHandler 구현

Developer-Groo committed:
feat: GlobalErrorCode 구현

Developer-Groo committed:
refctor: GlobalExceptionHandler 에 MessageSource 프로퍼티 제거 및 handleInternalServerError 메서드 추가

Developer-Groo committed:
feat: ValidationErrorCode 구현

- Validation 검증의 공통 메세지

Developer-Groo committed:
feat: ValidationErrorCodeResolver 구현

Developer-Groo committed:
refactor: 기존 메세지가 아닌 ErrorCode 사용을 위한 문자열 변경

Developer-Groo committed:
feat: Resolver 적용 및 핸들러 추가

- handleTypeMismatch()
- handleMissingRequestParam()

Developer-Groo committed:
feat: ErrorResponseDto 에 시간 필드 추가

- ZonedDateTime 활용

Developer-Groo committed:
Update TROUBLE_SHOOTING_TEMPLATE.md

Developer-Groo committed:
feat: handleInternalServerError 메서드에 로그 추가

taejunUM committed:
refactor: ArticleLikeController @Positive 어노테이션 추가

taejunUM committed:
style: 공백 추가

taejunUM committed:
refactor: parameter 변경

- dto네이밍 변경

taejunUM committed:
refactor: JPA 메서드 변경

taejunUM committed:
refactor: 메서드 리팩토링

-refactor: getArticleLikeCountAndLikedMembers 현재 좋아요 상태 반환
- dto 네이면 변경 반영
- repository JPA메서드 네이밍 변경 반영

taejunUM committed:
refactor: JPA메서드 네이밍 변경

taejunUM committed:
refactor: toggleLike 메서드 리팩토링

Developer-Groo committed:
refactor: ErrorResponseDto status 와 code 분리

Developer-Groo committed:
docs: TECHNOLOGY_DECISION_TEMPLATE 문서 수정

taejunUM committed:
refactor: 미사용 dto 삭제

# 본문은 아래에 작성

# 꼬릿말은 아래에 작성: ex) Github issue #23

# --- COMMIT END ---
# feat	    새로운 기능 추가
# fix	    버그 수정
# docs	    문서 변경 (README 등 문서 파일 수정)
# style	    코드 스타일 변경 (세미콜론, 들여쓰기, 공백 등)
# refactor  코드 리팩터링 (기능 변화 없음, 변수명 정리 등)
# test	    테스트 코드 추가 및 수정
# chore	    빌드 설정, 패키지 매니저 설정, 기타 중요하지 않은 작업
# ------------------
#   커밋 메시지는 영문 소문자로 시작
#   형식은 type: message 사용
#   하나의 커밋에는 하나의 목적만 작성
#   메시지는 명확하게 무엇을 했는지 설명
# ------------------
#   <Optional>
#   하위 커밋 설명이 필요한 경우, 본문(body)을 줄바꿈하여 추가 설명 작성
#   feat: support social login
#
#   Added Kakao and Google login support.
#   Refactored auth controller to handle provider-based login.

taejunUM committed:
style: 순서 변경
